tests/integration/telemetry/stats/prometheus/http/customizemetrics/customize_metrics_test.go

```diff
@@ -0,0 +1,188 @@
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/prometheus"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/util/retry"
+	util "istio.io/istio/tests/integration/telemetry"
+	promUtil "istio.io/istio/tests/integration/telemetry/stats/prometheus"
+	common "istio.io/istio/tests/integration/telemetry/stats/prometheus/http"
+)
+
+var (
+	client, server echo.Instance
+	appNsInst      namespace.Instance
+	promInst       prometheus.Instance
+)
+
+func TestCustomizeMetrics(t *testing.T) {
+	framework.NewTest(t).
+		Features("observability.telemetry.stats.prometheus.customize-metric").
+		Run(func(ctx framework.TestContext) {
+			destinationQuery := buildQuery()
+			retry.UntilSuccessOrFail(t, func() error {
+				if err := sendTraffic(t); err != nil {
+					t.Errorf("failed to send traffic")
+				}
+				if err := promUtil.QueryPrometheus(t, destinationQuery, promInst); err != nil {
+					t.Logf("prometheus values for istio_requests_total: \n%s", util.PromDump(promInst, "istio_requests_total"))
+					return err
+				}
+				return nil
+			}, retry.Delay(3*time.Second), retry.Timeout(90*time.Second))
+
+			promUtil.ValidateMetric(t, promInst, destinationQuery, "istio_requests_total", 1)
+		})
+}
+
+func TestMain(m *testing.M) {
+	framework.NewSuite(m).
+		Label(label.CustomSetup).
+		Setup(istio.Setup(common.GetIstioInstance(), setupConfig)).
+		Setup(testSetup).
+		Setup(setupEnvoyFilter).
+		Run()
+}
+
+func testSetup(ctx resource.Context) (err error) {
+	appNsInst, err = namespace.New(ctx, namespace.Config{
+		Prefix: "echo",
+		Inject: true,
+	})
+	if err != nil {
+		return
+	}
+
+	_, err = echoboot.NewBuilder(ctx).
+		With(&client, echo.Config{
+			Service:   "client",
+			Namespace: appNsInst,
+			Ports:     nil,
+			Subsets:   []echo.SubsetConfig{{}},
+		}).
+		With(&server, echo.Config{
+			Service:   "server",
+			Namespace: appNsInst,
+			Subsets:   []echo.SubsetConfig{{}},
+			Ports: []echo.Port{
+				{
+					Name:         "http",
+					Protocol:     protocol.HTTP,
+					InstancePort: 8090,
+				},
+			},
+		}).
+		Build()
+	if err != nil {
+		return err
+	}
+	promInst, err = prometheus.New(ctx, prometheus.Config{})
+	if err != nil {
+		return
+	}
+	return nil
+}
+
+func setupConfig(_ resource.Context, cfg *istio.Config) {
+	if cfg == nil {
+		return
+	}
+	cfg.ControlPlaneValues = `
+values:
+ telemetry:
+   v2:
+     prometheus:
+       configOverride:
+         inboundSidecar:
+           debug: false
+           stat_prefix: istio
+           metrics:
+           - name: requests_total
+             dimensions:
+               response_code: istio_responseClass
+               request_operation: istio_operationId
+`
+	cfg.Values["telemetry.v2.metadataExchange.wasmEnabled"] = "false"
```

Comment:
nit: these are defaults, I don't think you need to specify them here.

Reply:
removed